### ISSUE TITLE
Mvc.Analyzer test code exceeds MAX_PATH

### ISSUE THREAD

**Issue by HaoK** [dotnet/aspnetcore#9431]
Similar to https://github.com/aspnet/AspNetCore/issues/7882 

https://github.com/aspnet/AspNetCore/tree/master/src/Mvc/Mvc.Api.Analyzers/test/TestFiles and https://github.com/aspnet/AspNetCore/tree/master/src/Mvc/Mvc.Analyzers/test/TestFiles which causes helix to reject these work items since the combined path is too long

cc @dougbu

**Comment by dougbu** [comment]
@HaoK how urgent is this issue? I'm hoping I can do this next week, not this one.

**Comment by HaoK** [comment]
Not urgent at this point, whenever you can get to it is fine

**Comment by mkArtakMSFT** [comment]
@pranavkm reassigning this to you to handle in a future milestone.

**Comment by dougbu** [comment]
@mkArtakMSFT this is an `area-infrastructure` issue I was planning to handle next week.

@pranavkm let me know if you would like me to take it back.

/fyi @natemcmaster

**Comment by natemcmaster** [comment]
These really long file names are a continual source of grief for windows contributors who happen to clone our repo into anything other than something short like `C:\src\AspNetCore\`.

Would recommend shortening file names if possible. There isn't really another solution available.

**Comment by mkArtakMSFT** [comment]
Given this is a long file-name caused issue moving back to @dougbu!

**Comment by Pilchie** [comment]
@dougbu  - I didn't see this on your project backlog - is it happening for Preview 8 or 9?

**Comment by dougbu** [comment]
@Pilchie I am not sure why moving product files around landed back on my plate and missed @mkArtakMSFT's comment when it happened.

@mkArtakMSFT I don't have the bandwidth to come up with better / shorter names or to design a new approach in this test code. Please have someone more familiar with the analyzer tests do the needful.

**Comment by mkArtakMSFT** [comment]
Looking into this it seems we can simply shorten the file names for the test classes.
@ajaybhargavb let's try to handle this post preview9, given this is pure infrastructure change.

For example `AttributesShouldNotBeAppliedToPageModelAnalyzerTest/DiagnosticsAreReturned_IfAllowAnonymousIsAppliedToPageHandlerMethod.cs` path section only is really long. We can definitely rename those and make them much shorter.

**Comment by mkArtakMSFT** [comment]
@dougbu I guess (and I am not really sure now) that I was thinking this can be solved by enabling long-paths on our agents, and that's why this was assigned back to you. However, now I think that this will cause a similar issue on customer machines too (who would like to clone and run tests). So better to take action - so reassigned to Ajay.

**Comment by dougbu** [comment]
@pranavkm I think you'd be the right person to take this. @dotnet/aspnet-build doesn't know the innards of the Mvc.Api.Analyzers  and Mvc.Analyzers. Need to shorten the test asset names and that requires changing something in the tests themselves.

And, since CTI is hitting this, the urgency seems to have increased

**Comment by dougbu** [comment]
Cleared the milestone but we remain interested in seeing this done. As is, the errors due to long paths can be a barrier to entry for the community. @mkArtakMSFT does anyone else on your team have time to fixup these tests❔

**Comment by dougbu** [comment]
Should this be closed now that #38010 is in @pranavkm❔

@HaoK @dotnet/aspnet-build the remaining long paths are mostly in Microsoft.AspNetCore.Razor.Langage. The following are some of the longest relative paths in my local repo and the longest is 219 characters.

``` text
artifacts\obj\Microsoft.Extensions.Diagnostics.HealthChecks.EntityFrameworkCore.Tests\Debug\net7.0\Microsoft.Extensions.Diagnostics.HealthChecks.EntityFrameworkCore.Tests.GeneratedMSBuildEditorConfig.editorconfig
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\BindToComponent_EventCallback_SpecifiesValue_WithMatchingProperties\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\BindToComponent_SpecifiesValueAndChangeEvent_WithMatchingProperties\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\BindToComponent_SpecifiesValueAndChangeEvent_WithoutMatchingProperties\TestComponent.codegen.cs
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\BindToComponent_SpecifiesValueAndChangeEvent_WithoutMatchingProperties\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\ChildComponent_WithGenericChildContent_SetsParameterNameOnComponent\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\Component_WithEditorRequiredChildContent_ValueSpecifiedAsText_WithoutName\TestComponent.codegen.cs
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\Component_WithEditorRequiredChildContent_ValueSpecifiedAsText_WithoutName\TestComponent.ir.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\Component_WithEditorRequiredParameter_ValuesSpecifiedUsingSplatting\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\GenericComponent_GenericEventCallbackWithGenericTypeParameter_TypeInference\TestComponent.codegen.cs
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\GenericComponent_GenericEventCallbackWithGenericTypeParameter_TypeInference\TestComponent.ir.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentDesignTimeCodeGenerationTest\GenericComponent_GenericEventCallbackWithGenericTypeParameter_TypeInference\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentRuntimeCodeGenerationTest\BindToComponent_SpecifiesValueAndChangeEvent_WithoutMatchingProperties\TestComponent.mappings.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentRuntimeCodeGenerationTest\Component_WithEditorRequiredChildContent_ValueSpecifiedAsText_WithoutName\TestComponent.codegen.cs
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentRuntimeCodeGenerationTest\Component_WithEditorRequiredGroup_SomeValueSpecified_MultipleGroups\TestComponent.diagnostics.txt
src\Razor\Microsoft.AspNetCore.Razor.Language\test\TestFiles\IntegrationTests\ComponentRuntimeCodeGenerationTest\GenericComponent_GenericEventCallbackWithGenericTypeParameter_TypeInference\TestComponent.codegen.cs
```

**Comment by pranavkm** [comment]
Sure, I can close it. I had wanted to also tackle updating to use the Roslyn test infrastructure, but it's being tracked by https://github.com/dotnet/aspnetcore/issues/36300
